### PR TITLE
Interpolation tests & better skeleton test class

### PR DESCRIPTION
**Pull request by ButterscotchV**
Skeleton intepolation handling is a bit of a mess right now, this PR adds some extra unit tests and a new function for joint interpolation. No longer interferes with #1352.

Branched off from #1349 but without any of the actual functionality changes. This should be safe to merge at any time for additional testing and a cleaner unit test setup without breaking anything! :3

Also just a small note, we will likely remove the joint interpolation stuff entirely, but the function added here can be used to verify functionality in that regard, especially to ensure that we don't have functionality regression.

### CODE REVIEW

**Review by Eirenliel — APPROVED** [review]

File: server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt [hunk 1]
```diff
@@ -1736,11 +1736,48 @@ class HumanSkeleton(
...
+			val isLower = (ref * UP_ADJ).angleToR(extended) > FastMath.HALF_PI
+
+			return if (isBack && isLower) {
+				// Select longest rotation
+				if (extended.dot(ref) >= 0f) -extended else extended
+			} else {
+				// Select shortest rotation
+				extended.twinNearest(ref)
+			}
+		}
```
Comment (ImUrX) [inline server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt:1781]: maybe we should move this to `Quaternion`?

File: server/core/src/test/java/dev/slimevr/unit/TestTrackerSet.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+	/**
+	 * All the trackers in the set as a list.
+	 */
+	val setL = set.asList()
+
+	/**
+	 * All the trackers in the set plus the headset.
+	 */
+	val all = set.plus(head)
```
Comment (ImUrX) [inline server/core/src/test/java/dev/slimevr/unit/TestTrackerSet.kt:44]:
```suggestion
	val all = set + head
```

File: server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt [hunk 3]
```diff
@@ -58,4 +59,13 @@ object TrackerUtils {
...
 			FastMath.isApproxEqual(q1.z, q2.z, tolerance)
+
+	fun randFloat(random: Random, minInclusive: Float, maxExclusive: Float): Float = minInclusive + (random.nextFloat() * (maxExclusive - minInclusive))
+
+	fun randQuat(random: Random): Quaternion = Quaternion(
+		randFloat(random, -1f, 1f),
+		randFloat(random, -1f, 1f),
+		randFloat(random, -1f, 1f),
+		randFloat(random, -1f, 1f),
+	).unit()
```
Comment (ImUrX) [inline server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt:70]: just a comment but QuaternionTest has a guassian distribution-based random if interested

**Review by ImUrX — APPROVED** [review]

File: server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt (hunk 3 above)
Comment (ButterscotchV) [reply server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt:70]: I did see that, I figured this would be ideal to keep a seed per test rather than global? I'm not sure what is the best option

File: server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt (hunk 1 above)
Comment (ButterscotchV) [reply server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt:1781]: Hmm perhaps... This is very skeleton specific, but also that would make more logical sense, I think it's a good idea

Comment (ButterscotchV) [reply server/core/src/main/java/dev/slimevr/tracking/processor/skeleton/HumanSkeleton.kt]: I think this is good yeah? I also realized I can just use `twinNearest` directly lol this code is so much simpler

File: server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt (hunk 3 above)
Comment (Eirenliel) [reply server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt:70]: The best option is not to use randoms in tests, but to cover fixed edge cases.

**Review by Eirenliel — APPROVED** [review]

File: server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt (hunk 3 above)
Comment (ButterscotchV) [reply server/core/src/test/java/dev/slimevr/unit/TrackerTestUtils.kt]: Not randomized anymore :3

**Review by Eirenliel — APPROVED** [review]
